drivers/javascript/errors.coffee

```diff
@@ -24,9 +24,19 @@ class RqlServerError extends Error
 class RqlRuntimeError extends RqlServerError
 
 class RqlCompileError extends RqlServerError
-
 class RqlClientError extends RqlServerError
 
+class RqlInternalError extends RqlRuntimeError
+class RqlResourceError extends RqlRuntimeError
+class RqlLogicError extends RqlRuntimeError
+class RqlNonExistenceError extends RqlRuntimeError
```

Comment:
(This probably could be clearer in the protobuf, I tried to encode the hierarchy in the numbers used for the error types but it probably came out pretty opaque.)